cmd/util/util.go

```diff
@@ -165,3 +167,89 @@ func PromptUserForAccept(r io.Reader) bool {
+	//   all possible runtime.GOOS options: "android darwin dragonfly freebsd linux nacl netbsd openbsd plan9 solaris windows "
+
+	//386, arm, amd64 are only available k8s url options
+	// all possible runtime.ARCH options:  "386 amd64 amd64p32 arm arm64 ppc64 ppc64le mips mipsle mips64 mips64le mips64p32 mips64p32le ppc s390 s390x sparc sparc64 "
+
+	platform := runtime.GOOS
+	if platform != "windows" || platform != "darwin" {
+		platform = "linux" // not sure if I should do but it seems linux binary correct for most/all reasonable alternatives
+	}
+
+	verb := "run"
+	installInstructions := "curl -Lo kubectl https://storage.googleapis.com/kubernetes-release/release/%s/bin/%s/%s/kubectl && chmod +x kubectl && sudo mv kubectl /usr/local/bin/"
+	if runtime.GOOS == "windows" {
+		verb = "do"
+		installInstructions = "download kubectl from:\n" +
+			"https://storage.googleapis.com/kubernetes-release/release/%s/bin/%s/%s/kubectl\n" +
+			"Add kubectl to your system PATH"
+	}
+
+	latest, err := getLatestK8sTagVersion()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, err.Error())
+	}
+	_, err = exec.LookPath("kubectl")
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "========================================\n"+
+			"kubectl could not be found on your path.  kubectl is a requirement for using minikube\n"+
+			"To install kubectl, please %s the following:\n\n"+
+			installInstructions+"\n\n"+
+			"To disable this message, run the following:\n"+
+			"minikube config set WantKubectlDownloadMsg false\n"+
+			"========================================\n",
+			verb, latest, platform, runtime.GOARCH)
+
+		//Alternatively, minikube installs the binary:
+		// fmt.Println.("Would you like minikube to install kubectl") // Unsure if this is desired functionality
+		// if promptUser() .....
+		//    installKubectlp
+	}
+	// check if there is a newer version available of kubectl?
+	// I feel as though kubectl should do this?
+	// examine value of os.exec("kubectl","version")
+	// compare with getLatestK8sTagVersion()?
+}
+
+type K8sRelease struct {
+	Name string
+}
+
+type K8sReleases []K8sRelease
+
+func getJson(url string, target *K8sReleases) error {
+	r, err := http.Get(url)
+	if err != nil {
+		return errors.Wrap(err, "Error getting minikube version url via http")
+	}
+	defer r.Body.Close()
+
+	return json.NewDecoder(r.Body).Decode(target)
+}
+
+func getLatestK8sTagVersion() (string, error) {
+	k8sGithubURL := "https://api.github.com/repos/kubernetes/kubernetes/tags"
+	var releases K8sReleases
+	glog.Infof("Checking for updates...")
+	if err := getJson(k8sGithubURL, &releases); err != nil {
+		return "", errors.Wrap(err, "Error getting json from minikube version url")
+	}
+	if len(releases) == 0 {
+		return "", errors.Errorf("There were no json releases at the url specified: %s", k8sGithubURL)
+	}
+	// tags are sorted highest to lowest by semvar, thusly the first non-alpha/beta (no "-") will be the latest stable version
+	for _, release := range releases {
```

Comment:
Instead of this I think we should just download constants.DefaultKubernetesVersion of kubectl.

Asking a user to download kubectl 1.5.0 for their minikube running k8s 1.3 probably is going to lead to problems